pkg/codegen/dotnet/gen_program.go

```diff
@@ -772,3 +777,20 @@ func (g *generator) genNYI(w io.Writer, reason string, vs ...interface{}) {
 	})
 	g.Fgenf(w, "\"TODO: %s\"", fmt.Sprintf(reason, vs...))
 }
+
+func SetHighestPackageVersion(versionExpr *model.Expression, pkg string, packages []*schema.Package) {
```

Comment:
@AaronFriel this function should definitely be centralized instead of in each language's `utilities.go`, I tried putting it in the top-level codegen `utiliites.go` but was unable to import the required `hcl/model` package due to a circular dependency. 

Todo: add test